### PR TITLE
rework user serialization in session, persist sessions across restarts

### PR DESCRIPTION
**Pull request by JackVCurtis**

### CODE REVIEW

**Comment by Elkrival** [comment]
are we storing the entire user in the req.user session..?

File: server/app.js [hunk 1]
```diff
@@ -127,28 +126,48 @@ app.use(passport.initialize())
...
   new localStrategy(
     {
       ...PASSPORT_FIELDS_ATTRIBUTES,
-      passReqToCallback: true,
     },
-    localSignIn
-  )
-)
+    async (username, password, done) => {
+      const { appDb } = app.locals
```
Comment (Elkrival) [inline server/app.js:134]: 💪

File: server/app.js [hunk 2]
```diff
@@ -127,28 +126,48 @@ app.use(passport.initialize())
...
+        uid: 1,
+      })
+
+      if (!user) {
+        return done(null, false)
+      }
+
+      if (!verifyHash(password, user?.password)) {
+        return done(null, false)
+      }
```
Comment (Elkrival) [inline server/app.js]: nit
```suggestion

      if (!user) return done(null, false)
      

      if (!verifyHash(password, user?.password)) return done(null, false)
      
```

File: server/controllers/authController/index.js [hunk 3]
```diff
@@ -1,61 +1,88 @@
...
-            mail,
-            account_expires,
-            preferences: { config: configuration._id.toString() },
-          }
-          const newUser = await UserModel.create(appDb, newUserAttributes)
-          const registrationMailer = new RegistrationMailer(newUser)
+      async function (err, user, info) {
+        if (err) {
+          return next(err)
+        }
```
Comment (Elkrival) [inline server/controllers/authController/index.js]: nit
```suggestion
        if (err) return next(err)
```

File: server/controllers/authController/index.js [hunk 4]
```diff
@@ -1,61 +1,88 @@
...
-          await registrationMailer.sendMail()
+        const { uid } = user
 
-          return res.status(200).json({ data: newUser })
-        } catch (error) {
-          return res.status(400).json({ error: error.message })
+        const { appDb, dataDb } = req.app.locals
+
+        const userInfo = await UserModel.update(appDb, dataDb, uid, {
+          last_logon: Date.now(),
```
Comment (Elkrival) [inline server/controllers/authController/index.js:24]: Can we use dayjs to set this date with a timezone..?

File: server/utils/passport/ensureStudyPermission.js [hunk 5]
```diff
@@ -6,9 +6,7 @@ export default async function ensureStudyPermission(req, res, next) {
 
   try {
     const { study } = req.params
-    const { appDb } = req.app.locals
-    const user = await UserModel.findOne(appDb, { uid: req.user })
-    const { blocked, access } = user
+    const { blocked, access } = req.user
```
Comment (Elkrival) [inline server/utils/passport/ensureStudyPermission.js:11]: I think we need a user look up here in case the access to the study has changed

**Review by Elkrival — COMMENTED** [review]
It's looking good

File: server/controllers/authController/index.js (hunk 4 above)
Comment (JackVCurtis) [reply server/controllers/authController/index.js:24]: Eh, we could but this is the existing functionality and I'm hesitant to change it.

File: server/utils/passport/ensureStudyPermission.js (hunk 5 above)
Comment (JackVCurtis) [reply server/utils/passport/ensureStudyPermission.js:11]: 94abecf

File: server/controllers/authController/index.js (hunk 3 above)
Comment (JackVCurtis) [reply server/controllers/authController/index.js]: 618fe0b

File: server/app.js (hunk 2 above)
Comment (JackVCurtis) [reply server/app.js]: 618fe0b

**Review by Elkrival — APPROVED** [review]
👍

**Pull request merged**